app/models/topic.rb

```diff
@@ -32,8 +32,16 @@ def guide_content_ids
     guides.pluck(:content_id).uniq
   end
 
+  def add_section
+    self.topic_sections.build(position: next_position_in_list)
+  end
+
 private
 
+  def next_position_in_list
+    (TopicSection.belonging_to_topic(self.id).maximum(:position) || 0) + 1
+  end
```

Comment:
The self in `self.id` isn't necessary here. As it won't be able to find a local variable called `id` it will look for the method `#id` anyway.